### ISSUE TITLE
Calibration with respect to the car

### ISSUE THREAD

**Issue by Dcasadoherraez** [snail-radar/dataset_tools#5]
Hello again!
I was wondering if there's any plans of adding the calibration of the sensor set with respect to the SUV car.
It would be really helpful for autonomous driving applications :)

It's usually given w.r.t. the center of the rear axle as shown in the image below and it's provided in other datasets like the [View of Delft](https://intelligent-vehicles.org/datasets/view-of-delft/).

"Radar-Only Odometry and Mapping for Autonomous Vehicles"
![image](https://github.com/user-attachments/assets/8108e167-1267-4fe6-9c2f-b5e72b806797)

**Comment by JzHuai0108** [comment]
We can provide some rough values measured by hand, but since we do not log the wheel odometer, it cannot be refined by an optimizer.

**Comment by JzHuai0108** [comment]
Note the Body frame FLU at hesai XT32,
Define vehicle frame FLU at the center of the rear wheel axle, then
Body_t_vehicle = [-0.128, 0, -0.170] +/- [0.05, 0.05, 0.05] m
Body_R_vehicle = eye(3).
Body_T_vehicle = [R, t] transforms a point in the vehicle frame to the body frame.

**Comment by Dcasadoherraez** [comment]
Thanks so much!!! 
I really appreciate the effort :)
I am a bit skeptical about the Body_t_vehicle however. I'm not sure the picture illustrates only 17cm between the xt32 and the rear axle (at wheel height)
![image](https://github.com/user-attachments/assets/4c3bef74-57ae-4f15-acd5-a2a130cf316e)

**Comment by Dcasadoherraez** [comment]
The car to radar calibration provided by the authors didn't really work. Here are some sample values that are reasonably "okay-ish". However a more accurate calibration would be appreciated :)

```
        if date < 20231104:
            raise "No calibration available for dates < 20231104"
        elif date == 20231201:
            T_car_radar = np.array([[ 0.99975801  ,0.02199823  ,0.          ,1.2       ],
                                     [-0.02199823  ,0.99975801  ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])
        elif date == 20231208:
            T_car_radar = np.array([[ 0.99978988  ,0.02049856  ,0.          ,1.25      ],
                                     [-0.02049856  ,0.99978988  ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])
        elif date == 20231213:
            T_car_radar = np.array([[ 0.9998555   ,0.01699918  ,0.          ,1.25      ],
                                     [-0.01699918  ,0.9998555   ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])
        elif date == 20240115:
            T_car_radar = np.array([[ 0.9996219   ,0.02749653  ,0.          ,1.25      ],
                                     [-0.02749653  ,0.9996219   ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])
        elif date == 20240116:
            T_car_radar = np.array([[ 0.99966202  ,0.02599707  ,0.          ,1.27      ],
                                     [-0.02599707  ,0.99966202  ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])
        else:
            T_car_radar = np.array([[ 0.99960803  ,0.02799634  ,0.          ,1.3       ],
                                     [-0.02799634  ,0.99960803  ,0.          ,0.        ],
                                     [ 0.          ,0.          ,1.          ,0.        ],
                                     [ 0.          ,0.          ,0.          ,1.        ]])

        

        ```
